### ISSUE TITLE
[duckdb] Unable to Lex underscores in numeric literals

### ISSUE THREAD

**Issue by paultiq** [sqlfluff/sqlfluff#7375]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

With DuckDB dialect, SQLFluff fails to parse valid numeric literals that contain underscores: https://duckdb.org/docs/stable/sql/data_types/literal_types#underscores-in-numeric-literals

### Expected Behaviour

123_456 should be lexed and parsed properly, same as 123456

### Observed Behaviour

> L:   1 | P:   8 |  LXR | Unable to lex characters: '123_456'
> 

> L:   1 | P:   8 |  PRS | Line 1, Position 8: Found unparsable section: '123_456'

### How to reproduce

```sql
select 123_456
```

> uv run sqlfluff lint --dialect duckdb test1.sql
> == [test1.sql] FAIL
> L:   1 | P:   1 | CP01 | Keywords must be upper case.
>                        | [capitalisation.keywords]
> L:   1 | P:   8 |  LXR | Unable to lex characters: '123_456'
> L:   1 | P:   8 |  PRS | Line 1, Position 8: Found unparsable section: '123_456'
> WARNING: Parsing errors found and dialect is set to 'duckdb'. Have you configured your dialect correctly?
> All Finished 📜 🎉!

### Dialect

duckdb

### Version

4.0.0

### Configuration

[tool.sqlfluff.core]
dialect = "duckdb"
templater = "jinja"
sql_file_exts = ".sql,.sql.j2,.dml,.ddl"
exclude_rules = [
    "AM04",
    "ST06",
    "LT01",
    "LT12",
    "JJ01"
]

[tool.sqlfluff.indentation]
tab_space_size = 2

[tool.sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
